utils/sample_qc.py

```diff
@@ -364,98 +364,109 @@ def infer_sex(
-                (chr_y_ploidy < xx_y_ploidy_cutoff) &
-                (f_stat < f_stat_female_cutoff),
-                'XX')
-                .when(
-                (chr_x_ploidy > xy_x_ploidy_cutoffs[0]) &
-                (chr_x_ploidy < xy_x_ploidy_cutoffs[1]) &
-                (chr_y_ploidy > xy_y_ploidy_cutoffs[0]) &
-                (chr_y_ploidy < xy_y_ploidy_cutoffs[1]) &
-                (f_stat > f_stat_male_cutoff),
-                'XY')
+                    ((chr_x_ploidy > x_ploidy_cutoffs[0][0]) &
+                    (chr_x_ploidy < x_ploidy_cutoffs[0][1])),
+                    'X')
                 .when(
-                (chr_x_ploidy > xy_x_ploidy_cutoffs[0]) &
-                (chr_x_ploidy < xy_x_ploidy_cutoffs[1]) &
-                (chr_y_ploidy > yy_y_ploidy_cutoff) &
-                (f_stat > f_stat_male_cutoff),
-                'XYY')
+                    ((chr_x_ploidy > x_ploidy_cutoffs[1][0]) &
+                    (chr_x_ploidy < x_ploidy_cutoffs[1][1])),
+                    'XX')
                 .when(
-                (chr_x_ploidy > xx_x_ploidy_cutoffs[0]) &
-                (chr_x_ploidy < xx_x_ploidy_cutoffs[1]) &
-                (chr_y_ploidy > xy_y_ploidy_cutoffs[0]) &
-                (chr_y_ploidy < xy_y_ploidy_cutoffs[1]) &
-                (f_stat < f_stat_female_cutoff),
-                'XXY')
+                    (chr_x_ploidy >= x_ploidy_cutoffs[2]),
+                    'XXX')
+                .default('Ambiguous')
+        ),
+        Y_karyotype=(
+            hl.case()
+                .when(chr_y_ploidy < y_ploidy_cutoffs[0], '')
                 .when(
-                (chr_x_ploidy > xxx_x_ploidy_cutoff) &
-                (chr_y_ploidy < xx_y_ploidy_cutoff) &
-                (f_stat < f_stat_female_cutoff),
-                'XXX')
+                    ((chr_y_ploidy > y_ploidy_cutoffs[0]) &
+                    (chr_y_ploidy < y_ploidy_cutoffs[1])),
+                    'Y')
                 .when(
-                (chr_y_ploidy < xx_y_ploidy_cutoff) &
-                (chr_x_ploidy > xy_x_ploidy_cutoffs[0]) &
-                (chr_x_ploidy < xy_x_ploidy_cutoffs[1]) &
-                (f_stat > f_stat_male_cutoff),
-                'X0')
+                     chr_y_ploidy >= y_ploidy_cutoffs[2],
+                    'YY')
                 .default('Ambiguous')
-        )
+            )
     )
-
+    
     return sex_expr.annotate(
-        sex=(
-            hl.case()
-                .when(sex_expr.sex_karyotype == 'XX', 'female')
-                .when(sex_expr.sex_karyotype == 'XY', 'male')
-                .or_missing()
-        ),
-        is_female=(
+        sex_karyotype=(
             hl.case()
-                .when(sex_expr.sex_karyotype == 'XX', True)
-                .when(sex_expr.sex_karyotype == 'XY', False)
-                .or_missing()
-        )
+                .when((hl.len(sex_expr.X_karyotype) < 4) & (hl.len(sex_expr.Y_karyotype) < 3),
```

Comment:
Is the only condition being tested here that both `X_karyotype` and `Y_karyotype` are  not `Ambiguous`  ?
Maybe consider something like:
```
sex_karyotype=hl.cond(
    (sex_expr.X_karyotype == 'Ambiguous') |  (sex_expr.X_karyotype == 'Ambiguous'),
    'Ambiguous',
    sex_expr.X_karyotype + sex_expr.Y_karyotype 
)
```